File: src/main/java/com/moovie/backend/data/repository/CommentsRepositoryImpl.java
package com.moovie.backend.data.repository;

import com.moovie.backend.data.remote.comments.CommentsDataSource;
import com.moovie.backend.domain.repository.CommentsRepository;
import com.moovie.backend.domain.model.CommentListing;
import org.springframework.stereotype.Repository;

@Repository
public class CommentsRepositoryImpl implements CommentsRepository {

    private final CommentsDataSource commentsDataSource;

    public CommentsRepositoryImpl(CommentsDataSource commentsDataSource) {
        this.commentsDataSource = commentsDataSource;
    }

    @Override
    public CommentListing getComments(String contentId, int page, int pageSize) {
        return commentsDataSource.getComments(contentId, page, pageSize);
    }
}

File: src/main/kotlin/org/mobyle/data/repository/CommentsRepositoryImpl.kt
package org.mobyle.data.repository

import org.mobyle.data.remote.CommentsDataSource
import org.mobyle.domain.model.CommentListing
import org.mobyle.domain.repository.CommentsRepository

class CommentsRepositoryImpl(
    private val commentsDataSource: CommentsDataSource
) : CommentsRepository {

    override suspend fun getComments(contentId: String, page: Int, pageSize: Int): CommentListing {
        return commentsDataSource.getComments(contentId, page, pageSize)
    }
}
